File: hackerrank/problem_solving/algorithms/implementation/breaking_the_records.js
// https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem
"use strict";

const fs = require("fs");

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", inputStdin => {
    inputString += inputStdin;
});

process.stdin.on("end", function () {
    inputString = inputString
        .replace(/\s*$/, "")
        .split("\n")
        .map(str => str.replace(/\s*$/, ""));

    main();
});

function readLine() {
    return inputString[currentLine++];
}

// Complete the breakingRecords function below.
const breakingRecords = s => {
    let min = s[0],
        max = s[0],
        result = [0, 0];
    s = s.slice(1);
    for (let sc of s) {
        if (sc > max) {
            result[0]++;
            max = sc;
        } else if (sc < min) {
            result[1]++;
            min = sc;
        }
    }
    return result;
};
function main() {
    const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

    const n = parseInt(readLine(), 10);

    const scores = readLine()
        .split(" ")
        .map(scoresTemp => parseInt(scoresTemp, 10));

    const result = breakingRecords(scores);

    ws.write(result.join(" ") + "\n");

    ws.end();
}

File: hackerrank/problem_solving/algorithms/implementation/breaking_the_records.py
# https://www.hackerrank.com/challenges/breaking-best-and-worst-records/problem
input()
input_arr = [*map(int, input().split(' '))]


def solution(a):
    min_, max_, result = a[0], a[0], [0, 0]
    a = a[1:]
    for i in a:
        if(i > max_):
            result[0] += 1
            max_ = i
        elif(i < min_):
            result[1] += 1
            min_ = i
    return result


if __name__ == '__main__':
    print(' '.join(map(str, solution(input_arr))))
